File: experiments/C/framework_template.c
#include <omp.h>

#define NUM_ITERATIONS (10000)

void setup();
void postprocessing();
void run();

// ##########################################################
//        THIS SECTION IS FOR I/O VARIABLES DECLARATIONS
// ##########################################################

// ##########################################################

int main() {
    setup();
    run();
    postprocessing();
}

void setup() {

}

void run() {
    for(int i=0; i<NUM_ITERATIONS; i++) {
        double start_benchmark = omp_get_wtime();
        run_inference(in_noisy, in_hidden1, in_hidden2, output, out_hidden1, out_hidden2);
        double end_benchmark = omp_get_wtime();
        double time = end_benchmark - start_benchmark;
    }
}

void postprocessing() {
    
}

File: experiments/python/framework_template.py
import onnxruntime as ort
import numpy as np
import time

num_iterations = 100000

########################################################
#                PUT I/O VARIABLES HERE                #
########################################################

model_path = 'qmlp.onnx'
dummy_input = None
session = None
exp_collector = []

########################################################

def setup():
    session = ort.InferenceSession(model_path)
    input_shape = (1, 41)
    dummy_input = np.random.randn(*input_shape).astype(np.float32)

def run():
    input_name = session.get_inputs()[0].name
    for _ in range(num_iterations):
        start_time = time.time()
        session.run(None, {input_name: dummy_input})
        end_time = time.time()
        cpu_inference_time = end_time - start_time
        exp_collector.append(cpu_inference_time)

def postprocessing():
    average = sum(exp_collector) / num_iterations
    maxval = max(exp_collector)
    print("average: {average}")
    print("maximum: {maxval}")